sdk/python/feast/infra/offline_stores/redshift.py

```diff
@@ -153,16 +153,17 @@ def query_generator() -> Iterator[str]:
                 full_feature_names=full_feature_names,
             )
 
-            yield query
-
-            # Clean up the uploaded Redshift table
-            aws_utils.execute_redshift_statement(
-                redshift_client,
-                config.offline_store.cluster_id,
-                config.offline_store.database,
-                config.offline_store.user,
-                f"DROP TABLE {table_name}",
-            )
+            try:
+                yield query
+            finally:
+                # Always clean up the uploaded Redshift table
+                aws_utils.execute_redshift_statement(
+                    redshift_client,
+                    config.offline_store.cluster_id,
+                    config.offline_store.database,
+                    config.offline_store.user,
+                    f"DROP TABLE {table_name}",
```

Comment:
This should be some form of `DELETE IF EXISTS` too, but I don't think Redshift SQL supports that